### PR TITLE
Fix python dependencies

### PR DESCRIPTION
**Pull request by robin-reckmann**
openpilot doesnt compile with the latest agnos master. The following changes make it work again.
- Re-enable pyopencl
- Downgrade numpy
- Add future-fstrings

### CODE REVIEW

File: userspace/pyproject.toml [hunk 1]
```diff
@@ -39,7 +40,7 @@ dependencies = [
   "pydub",
   "pyjwt",
   "pylint",
-#"pyopencl == 2024.1", #FIXME: has some installation errors
+  "pyopencl == 2024.1",
```
Comment (adeebshihadeh) [inline userspace/pyproject.toml:43]: any idea why we have to pin? this is just a temp fix

File: userspace/pyproject.toml [hunk 2]
```diff
@@ -26,7 +27,7 @@ dependencies = [
   "libusb1",
   "markdown-it-py",
   "nose",
-  "numpy",
+  "numpy < 2.0.0",
```
Comment (adeebshihadeh) [inline userspace/pyproject.toml:30]:
```suggestion
  "numpy < 2.0.0",  # pending bumping in openpilot
```

File: userspace/pyproject.toml [hunk 3]
```diff
@@ -17,6 +17,7 @@ dependencies = [
   "evdev",
   "flake8",
   "flask",
+  "future-fstrings",
```
Comment (adeebshihadeh) [inline userspace/pyproject.toml:20]: what's this needed for?

Comment (robin-reckmann) [reply userspace/pyproject.toml:20]: acados

Comment (adeebshihadeh) [reply userspace/pyproject.toml:20]: can you leave a comment?

File: userspace/pyproject.toml (hunk 1 above)
Comment (robin-reckmann) [reply userspace/pyproject.toml:43]: Any newer version also needs a newer version of cmake to compile. Cmake >= 3.17 is not available via apt in 20.04. We could pull it in via pip.

Comment (adeebshihadeh) [reply userspace/pyproject.toml:43]: that's fine, let's just leave a comment

Comment (robin-reckmann) [reply userspace/pyproject.toml:43]: You mean keep it pinned for now? We can probably unpin it with 24.04 later.

Comment (adeebshihadeh) [reply userspace/pyproject.toml:43]: yeah, but leave a comment about why it's pinned

**Pull request merged**